pkg/testacc/2_check_destroy_test.go

```
@@ -0,0 +1,641 @@
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider/resources"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+)
+
+// TODO [next PRs]: this file is currently almost exact copy of check_destroy file in old acceptance package:
+// 	- the package name was changed
+//  - TestClient() -> testClient()
+
+func ComposeCheckDestroy(t *testing.T, resources ...resources.Resource) func(*terraform.State) error {
+	t.Helper()
+
+	return func(s *terraform.State) error {
+		errs := make([]error, 0)
+		for _, resource := range resources {
+			checkFunc := CheckDestroy(t, resource)
+			errs = append(errs, checkFunc(s))
+		}
+		return errors.Join(errs...)
+	}
+}
+
+func CheckDestroy(t *testing.T, resource resources.Resource) func(*terraform.State) error {
+	t.Helper()
+	// TODO [SNOW-1653619]: use TestClient() here
+	client := atc.client
+	t.Logf("running check destroy for resource %s", resource)
+
+	return func(s *terraform.State) error {
+		for _, rs := range s.RootModule().Resources {
+			if rs.Type != resource.String() {
+				continue
+			}
+			t.Logf("found resource %s in state", resource)
+			ctx := context.Background()
+			id, err := decodeSnowflakeId(rs, resource)
+			if err != nil {
+				return err
+			}
+			if id == nil {
+				return fmt.Errorf("could not get the id of %s", resource)
+			}
+			showById, ok := showByIdFunctions[resource]
+			if !ok {
+				return fmt.Errorf("unsupported show by id in cleanup for %s, with id %v", resource, id.FullyQualifiedName())
+			}
+			if showById(ctx, client, id) == nil {
+				return fmt.Errorf("%s %v still exists", resource, id.FullyQualifiedName())
+			} else {
+				t.Logf("resource %s (%v) was dropped successfully in Snowflake", resource, id.FullyQualifiedName())
+			}
+		}
+		return nil
+	}
+}
+
+func decodeSnowflakeId(rs *terraform.ResourceState, resource resources.Resource) (sdk.ObjectIdentifier, error) {
```

I know this is a copy, but this function and the above CheckDestroy seem incorrect with the identifiers that are parsed here and then passed to the showByIdFunctions. Currently, the resource identifiers are inconsistent as some of the resources still use `DecodeSnowflakeID`. It's also strange, because it should raise an error (or a warning) in this case, but either I'm wrong or this was missed somehow because it's only logged and not errors out.

---

Yeah, it's strange, I extracted it as SNOW-2110727

---

Addressed in #3967.